### PR TITLE
feat(cdk): expose authorizer id and authorization type

### LINKED ISSUE #31605: (aws-apigatewayv2-authorizers): Expose Authorizer ID to make it reusable from other projects
### Describe the feature

I need to be able to retrieve the Authorizer ID (the generated ID, not the one passed to the Construct) from the L2 construct.
Looking through the code I guess the issue is that the authorizer is not created until it's attached to a route. So I cannot personally create a PR for this since I don't know how references are resolved when the templates are generated.

### Use Case

I want to be able to reuse the authorizer in other projects in the same account.

### Proposed Solution

Expose the id the same way other constructs are.

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.159.0

### Environment details (OS name and version, etc.)

Windows 11 23H2

### PR DESCRIPTION
**Pull request by JonWallsten**
### Issue # (if applicable)

Closes #31605.

### Reason for this change

I need to access the authorizer id and type to be able to import the authorizer in another CDK project

### Description of changes

I have added a public readonly variable for the authorizationType that was previously a string in the return object from the bind function. 
I have also added a getter that will return the authorizerId after the first binding. If not bound yet it will throw an error.

### Description of how you validated changes

I have added unit tests for all changed files.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by JonWallsten** [comment]
@pahud

**Comment by pahud** [comment]
Thank you for your PR.

This PR will need at least one community review/approval before the flow can bump it to pending for maintainer review queue.

Please check CDK Community Reviews for more details
https://github.com/aws/aws-cdk/wiki/CDK-Community-PR-Reviews

**Comment by JonWallsten** [comment]
@pahud It seems like the contribution guide is not really meant for Windows users. The devcontainer is not working. And the command like `npx lerna run build --skip-nx-cache` fails becuase it relies on unix features (like chmod)  or `yarn test aws-apigatewayv2-authorizers` is not finding any test. I'm not really sure how to contribute if I cannot do it from a windows machine.

**Comment by pahud** [comment]
@JonWallsten My personal favorite is CodeCatalyst DevEnv.

You may find this article helpful FYI
https://community.aws/content/2jhmIXdVWZJy3uJnUfEzEmp652j/contributing-to-aws-cdk

**Comment by JonWallsten** [comment]
@pahud: I'm trying to understand how the integration tests works. I have added som code to the test and built the project. But I can't run the test due to 
```
Unable to resolve AWS account to use. It must be either configured when you define your CDK Stack, or through the environment
  FAILED     aws-apigatewayv2-authorizers/test/http/integ.lambda-integ.lambda (undefined/us-east-1) 2.932s
      Integration test failed: Error: Command exited with status 1```
      
Since I had to move the development to my personal computer where I don't have an account, I can't seem to run tests. Is there any workaround for this?

**Review by GavinZZ — DISMISSED** [review]

**Comment by GavinZZ** [comment]
@JonWallsten Did you have a chance to figure out the integration test issue? Given the previous comments, it seems that you don't have an AWS account to run the test? Is my understanding correct?

File: packages/aws-cdk-lib/aws-apigatewayv2-authorizers/README.md
Comment (JonWallsten) [reply packages/aws-cdk-lib/aws-apigatewayv2-authorizers/README.md:339]: My idea was to hit two birds with one stone. Show the export of the variable, which on it's own is pretty useless, and then how to use it. I don't attach any value to it, so if you have any other proposal, let me know and I will change it.

**Comment by JonWallsten** [comment]
> @JonWallsten Did you have a chance to figure out the integration test issue? Given the previous comments, it seems that you don't have an AWS account to run the test? Is my understanding correct?

Long story short. So far I've spent 30 minutes on the code and the tests, and 4 days trying to get the project to build and test.
First try: Windows as-is. Did not work at all. (Work computer)
Second try: Windows DevContainer using Windows Subsystem for Linux. Unbearable slow. (Work computer)
Third try: Mac as-is. Everything worked fine with build and test, until I got to integration tests since I don't have a personal AWS account.
Forth try: Windows Using AWS CodeCatalyst DevContainer. Got issues with memory, upped my billing plan and choose a better machine, and it mostly works besided running unit tests in VSCode, and some projects fails to build without any error message. But I could then build them one by one manually. So now I'm back to error message about the integration tests. Will try to add the credentials for my account today.

**Review by GavinZZ — APPROVED** [review]
Thanks for contributing!

**Comment by GavinZZ** [comment]
> > @JonWallsten Did you have a chance to figure out the integration test issue? Given the previous comments, it seems that you don't have an AWS account to run the test? Is my understanding correct?
> 
> @GavinZZ Long story short. So far I've spent 30 minutes on the code and the tests, and 4 days trying to get the project to build and test. First try: Windows as-is. Did not work at all. (Work computer) Second try: Windows DevContainer using Windows Subsystem for Linux. Unbearable slow. (Work computer) Third try: Mac as-is. Everything worked fine with build and test, until I got to integration tests since I don't have a personal AWS account. Forth try: Windows Using AWS CodeCatalyst DevContainer. Got issues with memory, upped my billing plan and choose a better machine, and it mostly works besided running unit tests in VSCode, and some projects fails to build without any error message. But I could then build them one by one manually. So now I'm back to error message about the integration tests. Will try to add the credentials for my account today.
> 
> Edit: I have added the credentials to the credential files and the account to the config file as the default profile. But it does not seem to work. Do I have to configure the integration tests to pick up the credentials/config?
> 
> I even tried to add the profile to manually through the cli: `yarn integ test/aws-apigatewayv2-authorizers/test/http/integ.lambda.js --update-on-failed --profiles test`
> 
> Edit 2: Got it to work by calling `aws sso login --profile test`
> 
> Edit 3: I still have a problem that the build fails after a while. image And I can't run `./scripts/run-rosetta.sh` without all projects being built. So now I have to build them one by one manually.

Wow, glad that you're finally able to make it build and work. Windows are known to have issues with CDK development unfortunately, and we really need to improve it for better experiences. I'll reach out internally to see if there's any items we could do to improve Windows user experience.

**Pull request merged**
